pkg/drive/drive.go

```diff
@@ -18,40 +18,82 @@ package drive
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"os"
 	"path/filepath"
 
 	"github.com/google/uuid"
 	directcsi "github.com/minio/direct-csi/pkg/apis/direct.csi.min.io/v1beta3"
 	"github.com/minio/direct-csi/pkg/clientset"
+	"github.com/minio/direct-csi/pkg/fs/xfs"
 	"github.com/minio/direct-csi/pkg/listener"
 	"github.com/minio/direct-csi/pkg/sys"
 	"github.com/minio/direct-csi/pkg/utils"
+	"golang.org/x/sys/unix"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/klog/v2"
 )
 
+func mountDevice(device, target string, flags []string) error {
+	if err := os.MkdirAll(target, 0777); err != nil {
+		return err
+	}
+
+	klog.V(3).InfoS("mounting device", "device", device, "target", target)
+	return sys.SafeMount(device, target, "xfs", flags, "prjquota")
+}
+
+func getDevice(major, minor uint32) (string, error) {
+	name, err := sys.GetDeviceName(major, minor)
+	if err != nil {
+		return "", err
+	}
+	return "/dev/" + name, nil
+}
+
+func mknod(fsuuid string, major, minor uint32) error {
+	path := filepath.Join(sys.DirectCSIDevRoot, fsuuid)
+
+	if err := os.Remove(path); err != nil && !errors.Is(err, os.ErrNotExist) {
+		return err
+	}
```

Comment:
question: will `os.Remove` throw a "device busy" if the device is mounted?

Reply:
can we mknod only if the (major, minor) of the device is different from what we expect? 

i see that, we call this function during drive formatting, in-case if the drive formatting fails due to any error, the listener will retry the format requests multiple times.

in such cases, we may delete and create the device files too many times.

Reply:
1. I haven't checked that. For regular files, `os.Remove()` succeeded even file is in use.
2. The better way is just do `Mknod()` and ignore `os.ErrExist` error.